nx/lib/nx.ex

```diff
@@ -5976,15 +5976,27 @@ defmodule Nx do
 
   ### Examples
 
-      iex> Nx.complex(Nx.tensor(1), Nx.tensor(2))
+      iex> Nx.complex(Nx.tensor(1, type: :f32), Nx.tensor(2, type: :f32))
+      #Nx.Tensor<
+        c64
+        1.0+2.0i
+      >
```

Comment:
I wouldn't change this example because it makes for less clarity on the docs. A separate test for this is welcome, though